Panic when generating dotnet SDK for pulumi-aws

Missed this on my earlier review but it looks like the [fix PR](https://github.com/pulumi/pulumi/pull/7451) is causing a panic in pulumi-aws when running `make build_dotnet` - I was able to replicate this locally as well. 
https://github.com/pulumi/pulumi/runs/3013744300#step:8:4796

Full trace:
```
panic: unexpected type <nil>

goroutine 1 [running]:
github.com/pulumi/pulumi/pkg/v3/codegen/dotnet.(*modContext).typeString(0xc00a0d01c0, 0x0, 0x0, 0x8813fc0, 0x6, 0xc000000001, 0x8815db4, 0x6)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/codegen/dotnet/gen.go:420 +0x1d90
github.com/pulumi/pulumi/pkg/v3/codegen/dotnet.(*modContext).unionTypeString(0xc00a0d01c0, 0xc00799fdc0, 0x8813fc0, 0x6, 0x1, 0xc00e000101, 0x8815db4)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/codegen/dotnet/gen.go:266 +0x1ba
github.com/pulumi/pulumi/pkg/v3/codegen/dotnet.(*modContext).typeString(0xc00a0d01c0, 0x9461c38, 0xc00799fdc0, 0x8813fc0, 0x6, 0x1, 0x10edece, 0xc0079a980f)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/codegen/dotnet/gen.go:398 +0x112d
github.com/pulumi/pulumi/pkg/v3/codegen/dotnet.(*modContext).typeString(0xc00a0d01c0, 0x9461b48, 0xc00ce8c560, 0x8813fc0, 0x6, 0x1, 0x2, 0x0)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/codegen/dotnet/gen.go:319 +0x127f
github.com/pulumi/pulumi/pkg/v3/codegen/dotnet.(*modContext).typeString(0xc00a0d01c0, 0x9461bc0, 0xc00ce8c570, 0x8813fc0, 0x6, 0xc00f000001, 0x10f8731, 0xc0079a9804)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/codegen/dotnet/gen.go:293 +0x651
github.com/pulumi/pulumi/pkg/v3/codegen/dotnet.(*plainType).genInputProperty(0xc00fca5e80, 0x93ef000, 0xc00ec75890, 0xc00ce9f570, 0xc003d3d4f8, 0x4)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/codegen/dotnet/gen.go:459 +0xef
github.com/pulumi/pulumi/pkg/v3/codegen/dotnet.(*plainType).genInputType(0xc00fca5e80, 0x93ef000, 0xc00ec75890, 0x1, 0x4, 0xc00eb7d3b0)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/codegen/dotnet/gen.go:587 +0x3f8
github.com/pulumi/pulumi/pkg/v3/codegen/dotnet.(*modContext).genResource(0xc00a0d01c0, 0x93ef000, 0xc00ec75890, 0xc00ce963c0, 0x5, 0x5)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/codegen/dotnet/gen.go:1038 +0x1b65
github.com/pulumi/pulumi/pkg/v3/codegen/dotnet.(*modContext).gen(0xc00a0d01c0, 0xc005dbf0b0, 0x0, 0x0)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/codegen/dotnet/gen.go:1653 +0x8b0
github.com/pulumi/pulumi/pkg/v3/codegen/dotnet.GeneratePackage(0x89058e2, 0x28, 0xc01a8eb180, 0x0, 0x0, 0x0, 0x0)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/codegen/dotnet/gen.go:2011 +0x399
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.Language.emitSDK(0x7ffeefbff54e, 0x6, 0xc01a8eb180, 0x951d2f0, 0xc0007c2370, 0x880ef6c, 0x3, 0x0, 0x0, 0x0, ...)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.4.0/pkg/tfgen/generate.go:142 +0x20d
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.(*Generator).Generate(0xc0016bc400, 0x3, 0x9406320)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.4.0/pkg/tfgen/generate.go:754 +0x3a8
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.newTFGenCmd.func1(0xc000575080, 0xc000a2f400, 0x1, 0x5, 0x0, 0x0)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.4.0/pkg/tfgen/main.go:101 +0x228
github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil.RunFunc.func1(0xc000575080, 0xc000a2f400, 0x1, 0x5, 0xc00169f948, 0xc00169f970)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/go/common/util/cmdutil/exit.go:96 +0x51
github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil.RunResultFunc.func1(0xc000575080, 0xc000a2f400, 0x1, 0x5)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.6.2-0.20210708031642-ccae7c80f8c9/go/common/util/cmdutil/exit.go:112 +0x6b
github.com/spf13/cobra.(*Command).execute(0xc000575080, 0xc000132130, 0x5, 0x5, 0xc000575080, 0xc000132130)
        /Users/komal/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc000575080, 0x3, 0x9406320, 0x26)
        /Users/komal/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/komal/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.Main(0x880ef6c, 0x3, 0x9406320, 0x26, 0x951d2f0, 0xc0007c2370, 0x880ef6c, 0x3, 0x0, 0x0, ...)
        /Users/komal/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.4.0/pkg/tfgen/main.go:33 +0xa9
main.main()
        /Users/komal/go/src/github.com/pulumi/pulumi-aws/provider/cmd/pulumi-tfgen-aws/main.go:24 +0xb8
make: *** [build_dotnet] Error 2
```

--- a/pkg/codegen/utilities_test.go
+++ b/pkg/codegen/utilities_test.go
@@ -15,8 +15,10 @@
 package codegen
 
 import (
-	"github.com/stretchr/testify/assert"
 	"testing"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/stretchr/testify/assert"
 )
 
 func TestStringSetContains(t *testing.T) {
@@ -41,4 +43,21 @@ func TestStringSetSubtract(t *testing.T) {
 	assert.Equal(t, set34, set1234.Subtract(set125))
 	assert.Equal(t, setEmpty, set1234.Subtract(set1234))
 	assert.Equal(t, set1234, set1234.Subtract(setEmpty))
+}
+
+func TestSimplifyInputUnion(t *testing.T) {
+	u1 := &schema.UnionType{
+		ElementTypes: []schema.Type{
+			&schema.InputType{ElementType: schema.StringType},
+			schema.NumberType,
+		},
+	}
+
+	u2 := SimplifyInputUnion(u1)
+	assert.Equal(t, &schema.UnionType{
+		ElementTypes: []schema.Type{
+			schema.StringType,
+			schema.NumberType,
+		},
+	}, u2)
 }

--- a/pkg/codegen/utilities_types.go
+++ b/pkg/codegen/utilities_types.go
@@ -47,6 +47,8 @@ func SimplifyInputUnion(t schema.Type) schema.Type {
 	for i, et := range union.ElementTypes {
 		if input, ok := et.(*schema.InputType); ok {
 			elements[i] = input.ElementType
+		} else {
+			elements[i] = et
 		}
 	}
 	return &schema.UnionType{

